src/common/snippets/include/snippets/lowered/linear_ir.hpp

```diff
@@ -70,7 +70,7 @@ class LinearIR {
     using exprReverseIt = container::reverse_iterator;
     using constExprReverseIt = container::const_reverse_iterator;
 
-    LinearIR(Config config = {});
+    LinearIR(Config config = {}, const std::shared_ptr<IShapeInferSnippetsFactory>& factory = {});
```

Comment:
I'd suggest to reorder parameters:
`LinearIR(const std::shared_ptr<IShapeInferSnippetsFactory>& factory, Config config = {});`

As it's done below with model.
To avoid saturations with empty factor for shape inference.

Or to set `std::make_shared<<IShapeInferSnippetsFactory>()` by default at least.

What do you think?

Reply:
Actually sometimes we don't need `factory` at all (I'd say in the most test scenarios: we have only a few lowered passes which perform shape inference inside). But if `IShapeInferSnippetsFactory` is set by default, we could meet problems with backend specific ops which can't be matched with shape infer instances by the default factory (e.g. `IShapeInferSnippetsFactory` knows nothing about `BrgemmCPU`)
To avoid code which will create LIR with `CPUShapeInferFactory` in each backend specific tests, I decided to set factory's default parameter to nullptr.

If you disagree with me, please let me know -- we will discuss the better solution :)

Reply:
Discussed offline: decided to leave it so since we don't use shape inference in lowered tests for now. When we test shape inference using LoweredTests often, we will add it if necessary